src/components/select/bl-select.ts

```diff
@@ -325,46 +373,109 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
         ></bl-button>`
       : "";
 
-    return html`<fieldset
+    const isSearchBarVisible = this.searchBar && this.opened;
+    const isMultipleWithSelection = this.multiple && this._selectedOptions.length > 0;
+
+    const isDividerShown = isSearchBarVisible || isMultipleWithSelection;
+
+    const searchMagIcon = html`<bl-icon
+      class="search-mag-icon"
+      name="search"
+      style="color: var(--bl-color-primary);font-size: var(--bl-font-size-s)"
```

Comment:
For these, we can consider not using the inline styling. Looks not ugly rn but if we use it everywhere, it will become surely fyi